retrofit-converters/gson/src/test/java/retrofit2/converter/gson/GsonConverterFactoryTest.java

```diff
@@ -78,7 +78,7 @@
   }
 
   interface Service {
-    @POST("/") Call<AnImplementation> anImplementation(@Body AnImplementation impl);
+    @POST("/") Call<AnImplementation> anImplementation(@Body(ignoreNull = false) AnImplementation impl);
```

Comment:
It's more about what to do when we receive `null` than whether `null` is accepted. I started out with only allow/disallow but it doesn't cover a lot of common cases.